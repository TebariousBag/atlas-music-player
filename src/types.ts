// api response types from provided api
export interface PlaylistSong {
  id: string;
  title: string;
  artist: string;
  genre: string;
  duration: number;
}

// song extends from playlist giving the cover art and link to song
export interface Song extends PlaylistSong {
  cover: string;
  song: string;
}

// optional, skipping for now
export interface Lyrics {
  lyrics: string;
}

// compnent prop types
export interface CoverArtProps {
  selectedSong?: Song | null;
}

export interface SongTitleProps {
  selectedSong?: Song | null;
}

export interface PlayControlsProps {
  playSpeed: number;
  canGoBack: boolean;
  canGoForward: boolean;
  onChangeSpeed: () => void;
  ongoBack: () => void;
  ongoForward: () => void;
}
