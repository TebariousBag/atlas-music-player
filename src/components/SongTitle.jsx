export default function SongTitle() {
  return (
    <div className="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left">
      <h2 class="text-french-blue my-2 text-4xl font-bold">Painted in Blue</h2>
      <p class="my-2 text-xl text-slate-500">Soul Canvas</p>
    </div>
  );
}
