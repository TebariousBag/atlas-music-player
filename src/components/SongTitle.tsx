import { SongTitleProps } from "../types";

export default function SongTitle({ selectedSong }: SongTitleProps) {
  return (
    <div className="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600">
      <h2 className="text-french-blue my-2 text-4xl font-bold dark:text-gray-300">
        {selectedSong?.title || "Song not selected"}
      </h2>
      <p className="my-2 text-xl text-slate-500 dark:text-gray-400">
        {selectedSong?.artist || "No artist selected"}
      </p>
    </div>
  );
}
