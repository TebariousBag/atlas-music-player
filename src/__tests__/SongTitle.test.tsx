import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import SongTitle from "../components/SongTitle";
import { Song } from "../types";

describe("SongTitle", () => {
  it("renders with placeholder text when no song is selected", () => {
    const { container } = render(<SongTitle selectedSong={null} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with placeholder text when song is undefined", () => {
    const { container } = render(<SongTitle selectedSong={undefined} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with song title and artist", () => {
    const song: Song = {
      id: "1",
      title: "Test Song",
      artist: "Test Artist",
      genre: "Test Genre",
      duration: 180,
      cover: "https://google.com/cover.jpg",
      song: "test-song.mp3",
    };

    const { container } = render(<SongTitle selectedSong={song} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with long title and artist", () => {
    const longSong: Song = {
      id: "2",
      title:
        "This is a very long song title that might wrap to multiple lines and test the layout",
      artist:
        "This is a very long artist name that might also wrap and test the layout",
      genre: "Test Genre",
      duration: 300,
      cover: "https://google.com/cover.jpg",
      song: "test-song.mp3",
    };

    const { container } = render(<SongTitle selectedSong={longSong} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with empty title and artist", () => {
    const emptySong: Song = {
      id: "3",
      title: "",
      artist: "",
      genre: "Test Genre",
      duration: 180,
      cover: "https://google.com/cover.jpg",
      song: "test-song.mp3",
    };

    const { container } = render(<SongTitle selectedSong={emptySong} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with special characters in title and artist", () => {
    const specialSong: Song = {
      id: "4",
      title: 'Song with "quotes" & symbols!',
      artist: "Artist with Ã©mojis ðŸŽµ & symbols",
      genre: "Test Genre",
      duration: 180,
      cover: "https://google.com/cover.jpg",
      song: "test-song.mp3",
    };

    const { container } = render(<SongTitle selectedSong={specialSong} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
