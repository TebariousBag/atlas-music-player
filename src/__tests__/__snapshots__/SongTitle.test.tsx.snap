// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > renders with empty title and artist 1`] = `
<div
  class="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600"
>
  <h2
    class="text-french-blue my-2 text-4xl font-bold dark:text-gray-300"
  >
    Song not selected
  </h2>
  <p
    class="my-2 text-xl text-slate-500 dark:text-gray-400"
  >
    No artist selected
  </p>
</div>
`;

exports[`SongTitle > renders with long title and artist 1`] = `
<div
  class="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600"
>
  <h2
    class="text-french-blue my-2 text-4xl font-bold dark:text-gray-300"
  >
    This is a very long song title that might wrap to multiple lines and test the layout
  </h2>
  <p
    class="my-2 text-xl text-slate-500 dark:text-gray-400"
  >
    This is a very long artist name that might also wrap and test the layout
  </p>
</div>
`;

exports[`SongTitle > renders with placeholder text when no song is selected 1`] = `
<div
  class="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600"
>
  <h2
    class="text-french-blue my-2 text-4xl font-bold dark:text-gray-300"
  >
    Song not selected
  </h2>
  <p
    class="my-2 text-xl text-slate-500 dark:text-gray-400"
  >
    No artist selected
  </p>
</div>
`;

exports[`SongTitle > renders with placeholder text when song is undefined 1`] = `
<div
  class="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600"
>
  <h2
    class="text-french-blue my-2 text-4xl font-bold dark:text-gray-300"
  >
    Song not selected
  </h2>
  <p
    class="my-2 text-xl text-slate-500 dark:text-gray-400"
  >
    No artist selected
  </p>
</div>
`;

exports[`SongTitle > renders with song title and artist 1`] = `
<div
  class="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600"
>
  <h2
    class="text-french-blue my-2 text-4xl font-bold dark:text-gray-300"
  >
    Test Song
  </h2>
  <p
    class="my-2 text-xl text-slate-500 dark:text-gray-400"
  >
    Test Artist
  </p>
</div>
`;

exports[`SongTitle > renders with special characters in title and artist 1`] = `
<div
  class="border-aero my-6 flex flex-col rounded-xl border-4 bg-blue-200 p-4 text-left dark:border-gray-500 dark:bg-gray-600"
>
  <h2
    class="text-french-blue my-2 text-4xl font-bold dark:text-gray-300"
  >
    Song with "quotes" & symbols!
  </h2>
  <p
    class="my-2 text-xl text-slate-500 dark:text-gray-400"
  >
    Artist with Ã©mojis ðŸŽµ & symbols
  </p>
</div>
`;
